src/wovnio/wovnphp/Url.php

```diff
@@ -174,8 +183,45 @@ public static function removeLangCode($uri, $pattern, $lang_code, $lang_param_na
             case 'path':
             default:
                 // limit to one replacement
-                return preg_replace('/\/' . $lang_code . '(\/|$)/i', '/', $uri, 1);
+                $prefix = empty($site_prefix_path) ? '' : '/' . $site_prefix_path;
+                return preg_replace("@$prefix/$lang_code(/|$)@i", "$prefix/", $uri, 1);
                 break;
         }
     }
+
+    public static function shouldIgnoreBySitePrefixPath($uri, $settings)
```

Comment:
I think it would be better if you move this to Utils.php as a private function used by isIgnoredPath + move the tests. We should only be calling `isIgnoredPath`

Reply:
`shouldIgnoreBySitePrefixPath` uses `generateUrlRegex` method which is defined at Url.php.
I didn't want to make `generateUrlRegex` public.
And I think `generateUrlRegex` should be in `Url.php`.

Reply:
ok it's not too important as long as we only use isIgnoredPath. Could move either everything to Url.php or to Utils.php